### ISSUE TITLE
Operator shows some warnings on ARM (k3s) deployments

### ISSUE THREAD

**Issue by amarts** [kadalu/kadalu#276]
```
[2020-07-08 19:45:53,337] INFO [main - 520:deploy_storage_class] - Deployed StorageClass         manifest=/kadalu/templates/storageclass-kadalu.yaml
/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py:986: InsecureRequestWarning: Unverified HTTPS request is being made to host '10.43.0.1'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/response.py", line 697, in _update_chunk_length
    self.chunk_left = int(line, 16)
ValueError: invalid literal for int() with base 16: b''
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/response.py", line 437, in _error_catcher
    yield
  File "/usr/local/lib/python3.7/site-packages/urllib3/response.py", line 764, in read_chunked
    self._update_chunk_length()
  File "/usr/local/lib/python3.7/site-packages/urllib3/response.py", line 701, in _update_chunk_length
    raise httplib.IncompleteRead(line)
http.client.IncompleteRead: IncompleteRead(0 bytes read)
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/kadalu/main.py", line 556, in <module>
    main()
  File "/kadalu/main.py", line 551, in main
    crd_watch(core_v1_client, k8s_client)
  File "/kadalu/main.py", line 413, in crd_watch
    resource_version=resource_version):
  File "/usr/local/lib/python3.7/site-packages/kubernetes/watch/watch.py", line 144, in stream
    for line in iter_resp_lines(resp):
  File "/usr/local/lib/python3.7/site-packages/kubernetes/watch/watch.py", line 46, in iter_resp_lines
    for seg in resp.read_chunked(decode_content=False):
  File "/usr/local/lib/python3.7/site-packages/urllib3/response.py", line 793, in read_chunked
    self._original_response.close()
  File "/usr/lib64/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.7/site-packages/urllib3/response.py", line 455, in _error_catcher
    raise ProtocolError("Connection broken: %r" % e, e)
urllib3.exceptions.ProtocolError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))
```

**Comment by FlexibleToast** [comment]
Adding more info

It appears that kadalu successfully deployed.
```
kubectl get pods -n kadalu
NAME                        READY   STATUS    RESTARTS   AGE
csi-nodeplugin-bzzkf        3/3     Running   0          54m
csi-nodeplugin-mmstn        3/3     Running   0          54m
csi-nodeplugin-twkzn        3/3     Running   0          54m
csi-nodeplugin-xkg2t        3/3     Running   0          54m
csi-provisioner-0           4/4     Running   0          54m
operator-77b8f46bd6-hglns   1/1     Running   0          56m
```

However, when attempting to deploy a storage pod
```
---
apiVersion: kadalu-operator.storage/v1alpha1
kind: KadaluStorage
metadata:
  name: test-0a
spec:
  type: Replica1
  storage:
    - node: pikubed-m0
      device: /dev/sda
```

The storage pod isn't created. That's when I pulled the log available in the original post.

**Comment by amarts** [comment]
Looks like the traceback issue is a glitch in urllib3 version:

* https://github.com/psf/requests/issues/4248 
* https://github.com/urllib3/urllib3/issues/1516 (Looks like fix went in as recently as a month back).

Rechecked the operator logs, and it is sure that it didn't get any storage-deployment requests. I will further check on this and update the thread!

**Comment by aravindavk** [comment]
I see the Python urllib3 fix is part of 1.25.9 release. Rebuilding container images without any change may help to fix the issue.

https://github.com/urllib3/urllib3/compare/1.25.9...master (Pull request: https://github.com/urllib3/urllib3/pull/1888)

**Comment by amarts** [comment]
From latest build:

```
Downloading https://files.pythonhosted.org/packages/e1/e5/df302e8017440f111c11cc41a6b432838672f5a70aa29227bf58149dc72f/urllib3-1.25.9-py2.py3-none-any.whl (126kB)
```
It should be available in next 1hr, with that we can see if the issue gets fixed.

**Comment by FlexibleToast** [comment]
Just tried rebuilding the cluster. Has the same error. I also confirmed that the Image ID matched the latest arm64 image here https://hub.docker.com/layers/kadalu/kadalu-operator/master-arm/images/sha256-f88f8fba78e51a300a3d4c71226ae7570d5b3710bdc190f59bfe439ecb2b849f?context=explore.

**Comment by FlexibleToast** [comment]
I think I've narrowed down the issue. I was deploying a highly available k3s cluster using an etcd database. That etcd database was also hosted on the three Raspberry Pi 4 master nodes. I attempted an install using an external MariaDB on a completely different machine while keeping all other parts of the install were left the same. This time kadalu was able to deploy and did not have the Python errors. This issue seems to be related to the clearly unsupported use case of running etcd on the master nodes on top of arm. The k3s projects is expected to release a lightweight embedded etcd with their next version some time in August. https://github.com/rancher/k3s/pull/1770#issuecomment-640132960. I can retest with the new version of k3s if the embedded etcd capabilities are included when it is released.

It appears that a kadalu still isn't creating a PV when given a PVC though. PVC stays in Pending status. Will have to create a new issue for this.

**Comment by amarts** [comment]
> It appears that a kadalu still isn't creating a PV when given a PVC though. PVC stays in Pending status. Will have to create a new issue for this.

With #262 merged, we should be good on ARM. Can you please check `master` label deployment again? (not `master-arm`).

**Comment by FlexibleToast** [comment]
I thought I had this particular issue resolved with switching to MariaDB instead of a underprovisioned etcd. However, I used the external MariaDB I confirmed to work previously and just got the exact same error that started this issue and the storage pod doesn't deploy. I did it twice to make sure it wasn't some connection issue with the MariaDB that caused it.

Just to confirm, I deployed it with:
`kubectl create -f https://raw.githubusercontent.com/kadalu/kadalu/master/manifests/kadalu-operator-master.yaml`
